tools/build_defs/pkg/make_deb.py

```diff
@@ -142,7 +142,7 @@ def MakeDebianControlField(name, value, wrap=False):
                            break_long_words=False)
   else:
     result += value
-  return result.replace('\n', '\n ') + '\n'
+  return result.decode('utf-8').replace('\n', '\n ') + '\n'
```

Comment:
The decode() could be at line 295 
     return flagvalue.decode('utf-8')

But also we might want to add it at line 289
    flagvalue = flagvalue.decode('utf-8')
so that an @file usage can have a utf-8 path

Reply:
Sounds good. I'll play around with this a little. I had a friend suggest setting `changesdata = u''` and `changesfile = u''` as well so they won't have to be overridden at each access point.

Reply:
Doing some testing - it seems that doing:
```python
if flagvalue:
  flagvalue = flagvalue.decode('utf-8')
  ...
```
and removing the decode in: `return result.decode('utf-8').replace(...) + '\n\'` on line 145 will cause UnicodeDecodeErrors if one of the fields that doesn't use GetFlagValue(s) in CreateDeb (line 306) contains a utf-8 variable.

I could move the `flagvalue = flagvalue.decode('utf-8')` line further into the if statement nest so the `@file` directive will get it and leave the rest the same.

Reply:
I think that is worth it. We have other outstanding bugs for not supporting UTF-8 file paths. So GetFlagValue needs to do that. AND the content should be interpreted as utf-8 encoded.

But I think I have the solution to what you observed.  We upgrade str to unicode str in MakeDebianControlField
I changed one of the non GetFlagValue() attributes to be unicode also, so the test would have a that case too.
     maintainer = "soméone@somewhere.com",

If you have extra tests, I would like to try them against my suggestion. If you think that change works. I also have it patched on top of your PR, so I could to the merge of yours.

Full diff: https://github.com/bazelbuild/bazel/compare/master...aiuto:deb

inline diff

*** 133,148 ****
  def MakeDebianControlField(name, value, wrap=False):
    """Add a field to a debian control file."""
    result = name + ': '
    if type(value) is list:
!     value = ', '.join(value)
    if wrap:
!     result += ' '.join(value.split('\n'))
      result = textwrap.fill(result,
                             break_on_hyphens=False,
                             break_long_words=False)
    else:
      result += value
!   return result.decode('utf-8').replace('\n', '\n ') + '\n'
  
  
  def CreateDebControl(extrafiles=None, **kwargs):
--- 133,150 ----
  def MakeDebianControlField(name, value, wrap=False):
    """Add a field to a debian control file."""
    result = name + ': '
+   if type(value) is str:
+     value = value.decode('utf-8')
    if type(value) is list:
!     value = u', '.join(value)
    if wrap:
!     result += u' '.join(value.split('\n'))
      result = textwrap.fill(result,
                             break_on_hyphens=False,
                             break_long_words=False)
    else:
      result += value
!   return result.replace('\n', '\n ') + '\n'
  
  
  def CreateDebControl(extrafiles=None, **kwargs):
***************
*** 295,303 ****
  
  def GetFlagValue(flagvalue, strip=True):
    if flagvalue:
      if flagvalue[0] == '@':
        with open(flagvalue[1:], 'r') as f:
!         flagvalue = f.read()
      if strip:
        return flagvalue.strip()
    return flagvalue
--- 297,306 ----
  
  def GetFlagValue(flagvalue, strip=True):
    if flagvalue:
+     flagvalue = flagvalue.decode('utf-8')
      if flagvalue[0] == '@':
        with open(flagvalue[1:], 'r') as f:
!         flagvalue = f.read().decode('utf-8')
      if strip:
        return flagvalue.strip()
    return flagvalue

Reply:
Sounds good - if there are other outstanding utf-8 filepath bugs it'd be worth it to deal with that while we're here. 

Amusingly - that was the exact test case I was using. I'd be fine with you merging on top of my commits. I guess I didn't go far enough in changing things to unicode strings!

The one other thing I was noticing was that if the controlfile isn't utf-8 encoded when determining the size content will be dropped from it. I've been using `expect_log "Built-Using: Bazel"` to test for that as that seems to usually be the last line in the controlfile. However, I'm not sure how robust of a test that is.

Reply:
Yes. I saw your fix for that. I'll try to commit this now. Having flakiness with the windows/jdk8 tests. So this may take a few tries to catch a clean build.

Reply:
Looks like it passed. Thanks for the review!